docs/threat-model/README.md

```diff
@@ -0,0 +1,426 @@
+
+    vso <--> vault
+    vso <--> apiserver
+    vso<-->vso-cache
+    vaultconn-default-->vso
+    vaultauth-default-->vso
+    vaultconn-server2-->vso
+    vaultauth-app1-->vso
+    vaultauth-approle-->vso
+    vss --> vso
+    vds --> vso
+    vps --> vso
+    vss -.- k8s-secret-1
+    vds -.- k8s-secret-2
+    vps -.- k8s-secret-3
+    apiserver --> k8s-secret-1
+    apiserver --> k8s-secret-2
+    apiserver --> k8s-secret-3
+    k8s-secret-1 --> app1
+    k8s-secret-2 --> app1
+    k8s-secret-3 --> app1
+    k8s-secret-3 --> app2
+    apiserver<-->etcd
+    vso <--> vault-ext
+
+    classDef trustboundary stroke-dasharray: 4
+
+    class k8s-sg trustboundary
+    class vso-sg trustboundary
+    class workload-sg trustboundary
+    class vault-sg trustboundary
+    class cp-sg trustboundary
+    class k8s-sg trustboundary
+
+```
+
+## Threats
+
+The following threats were enumerated while considering several types of attacks that may affect the confidentiality or integrity of secrets synced from Vault:
+
+1. Secret material in transit between Vault, the Operator, and the API server.
+1. Secret material at rest in etcd.
+1. API access to Operator namespace (privileged service account, Kubernetes Secrets used as cache).
+1. Access to Operator pods (in-memory cache secrets, privileged service account).
+1. API access to workload namespace (workload Secrets, execing into containers, deploying pods that reference the created secrets).
+1. Access to workload secrets mounted into pods (ephemeral volumes, environment variables, in-memory values).
+1. Misconfiguration of Kubernetes RBAC and Operator custom resources.
+
+<table>
+  <tr>
+   <td>
+<strong>ID</strong>
+   </td>
+   <td><strong>Threat</strong>
+   </td>
+   <td><strong>Categories</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+   <td><strong>Risk</strong>
+   </td>
+   <td><strong>Mitigation</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>1
+   </td>
+   <td>An attacker can snoop on or tamper with secret data in transit between the Operator and Vault
+   </td>
+   <td>Information disclosure, tampering, integrity, spoofing
+   </td>
+   <td>The network route configured between the Operator and Vault could be within the cluster, or to an external address either over a VPC or the public internet.
+   </td>
+   <td>Likelihood: Moderate
+<p>
+Impact: High
+   </td>
+   <td>
+<ul>
+
+<li>Communication with Vault should be encrypted via TLS
+</li>
+</ul>
+   </td>
+  </tr>
+  <tr>
+   <td>2
+   </td>
+   <td>An attacker can snoop on or tamper with secret data in transit between the API Server and other components
+   </td>
+   <td>Information disclosure, tampering, integrity, spoofing
+   </td>
+   <td>The Kubernetes API server receives unencrypted Vault secret material from the Operator. It sends the same unencrypted material to the node’s kubelet when mounting secrets into pods, and directly to pods when they request it from the API.
+   </td>
+   <td>Likelihood: Moderate
```

Comment:
nit: When the README is actually rendered likelihood and impact comes out sticking too close

perhaps making the values italic+bold with help with better reading

Reply:
Thanks, I tested out with just bold and it looked ok to me, LMK what you think